internal/pkg/manifest/validate.go

```diff
@@ -1937,36 +1959,56 @@ func populateNLBPortsAndValidate(containerNameFor map[uint16]string, opts valida
 		return nil
 	}
 	nlb := opts.nlb
-	if nlb.Port == nil {
+	if nlb.PrimaryRoutingRule.Port == nil {
 		return nil
 	}
-	nlbPort, _, err := ParsePortMapping(nlb.Port)
+
+	if err := populateAndValidateNLBPorts(nlb.PrimaryRoutingRule, containerNameFor, opts); err != nil {
```

Comment:
is it a valid use case to specify only additional rules without specifying primary rule for target container etc.?

Reply:
No it is not a valid use case. The reason being what we discussed in the earlier [comment](https://github.com/aws/copilot-cli/pull/4595#discussion_r1129977121).